index.js

```diff
@@ -48,7 +58,14 @@ async function main(adapterAddress, startBlock = 0, endBlock = Infinity) {
     if (to !== ethers.constants.AddressZero) {
       if (scores[to]) {
         const { prevCheckpoint, score } = scores[to];
-        scores[to].score = score.plus(prevCheckpoint.amount.times(block - prevCheckpoint.block));
+        if (startBlock > prevCheckpoint.block) {
+          if (startBlock < block) {
+            scores[to].score = score.plus(prevCheckpoint.amount.times(block - startBlock));
+          }
+        } else {
+          scores[to].score = score.plus(prevCheckpoint.amount.times(block - prevCheckpoint.block));
+        }
+
         scores[to].prevCheckpoint = { amount: prevCheckpoint.amount.plus(amount), block };
       } else {
         scores[to] = { score: new Decimal(0), prevCheckpoint: { amount, block } };
```

Comment:
I find it a bit odd that `startBlock` only matters after we process our first block :) 
I guess this stems from my general confusion as to what startBlock is supposed to represent!

Reply:
Yep, think it's a big confusing overall, hope to clear this up with some comments!

Reply:
thanks they're very helpful!!